contracts/WETH10.sol

```diff
@@ -39,29 +40,22 @@ contract WETH10 {
                 address(this)));
     }
 
-    modifier lock() {
-        require(unlocked == 1, "locked");
-        unlocked = 0;
-        _;
-        unlocked = 1;
-    }    
-
-    receive() external payable lock {
+    receive() external payable {
         balanceOf[msg.sender] += msg.value;
         emit Transfer(address(0), msg.sender, msg.value);
     }
     
-    function deposit() external payable lock {
+    function deposit() external payable {
         balanceOf[msg.sender] += msg.value;
         emit Transfer(address(0), msg.sender, msg.value);
     }
     
-    function depositTo(address to) external payable lock {
+    function depositTo(address to) external payable {
         balanceOf[to] += msg.value;
         emit Transfer(address(0), to, msg.value);
     }
     
-    function withdraw(uint256 value) external lock {
```

Comment:
This is the discussion I really wanted to have. How safe is it to allow flash minting ETH?

1. The actual limit for Weth flashMint + withdraw is the ETH balance of Weth.sol, so it can't be used to break contracts by overflow.
2. On the other hand, any contract that accepts ETH, meaning any contract deployed so far, can be interacted with.

Initially I was quite scared about allowing `withdraw` during a flash mint, and hence the `lock`, but I've been thinking a lot about it and I can't see how it can be used for breaking other contracts that wouldn't be possible with, for example, a flash mint of Dai.

Reply:
Ya I can't think of any reason why flashMinting ETH would be a problem.

Aave and dYdX already allow flashLoaning ETH, right?